examples/Button.loading.jsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { Button } from '@itwin/itwinui-react';
+
+export default function App() {
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  return (
+    <>
+      <Button
+        loading={isLoading}
+        onClick={async () => {
+          ReactDOM.flushSync(() => setIsLoading(true));
```

Comment:
nit: Is the `flushSync` needed here?

Reply:
Yes, we want to show the loading state immediately before performing the action.

Reply:
You can see similar code here: https://github.com/iTwin/iTwinUI-migration-tool/blob/32924f564041761236002166b500f9a3f36b6170/app/(routes)/v3/preview/RunCodemodsInAllFilesButton.tsx#L31-L33

Reply:
To confirm, do you mean that showing the loading state in the next frame after the action has started has some issue?

Reply:
Yes. When you write code like this, React will batch all state updates, so you might not even see the loading state:

```jsx
setIsLoading(true);
doSomething();
setIsLoading(false);
```